datev_export_xml/models/datev_xml_generator.py

```diff
@@ -23,16 +23,23 @@ class DatevXmlGenerator(models.AbstractModel):
     _description = "DATEV XML Generator"
 
     @api.model
-    def check_xml_file(self, doc_name, root, xsd=None):
+    def check_xml_file(self, doc_name, root, xsd=None, invoice=None):
         if not xsd:
             xsd = "Document_v050.xsd"
 
         schema = tools.file_open(f"datev_export_xml/xsd_files/{xsd}")
         try:
+            if invoice:
+                invoice.datev_validation = False
+
             schema = etree.XMLSchema(etree.parse(schema))
             schema.assertValid(root)
         except (etree.DocumentInvalid, etree.XMLSyntaxError) as e:
             _logger.warning(etree.tostring(root))
+
+            if invoice:
+                invoice.datev_validation = str(e)
```

Comment:
This data is not going to be written as the later raise rollbacks the transaction.

Reply:
This is correct for the normal export but not for the validation feature (`action_validate`)